File: Java/PatchingArray.java
class Solution {
    public int minPatches(int[] nums, int n) {
        long missing = 1;
        int patches = 0; 
        int i = 0; 
        while (missing <= n) {
            if (i < nums.length && nums[i] <= missing) {
                missing += nums[i];
                i++;
            } else {
                patches++;
                missing += missing;
            }
        }

        return patches;
    }
}

File: Python/PatchingArray.py
class Solution:
    def minPatches(self, nums, n):
        index = changes = 0 
        comp = 1 

        while comp <= n:
            if index < len(nums) and nums[index] <= comp:
                comp += nums[index]
                index += 1
            else:
                comp += comp
                changes += 1

        return changes
